service/history/visibility_queue_task_executor.go

```diff
@@ -318,11 +318,30 @@ func (t *visibilityQueueTaskExecutor) processDeleteExecution(
 	ctx, cancel := context.WithTimeout(ctx, taskTimeout)
 	defer cancel()
 
+	weContext, release, err := getWorkflowExecutionContextForTask(ctx, t.shardContext, t.cache, task)
+	if err != nil {
+		return err
+	}
+	defer func() { release(retError) }()
+
+	mutableState, err := weContext.LoadMutableState(ctx, t.shardContext)
+	if err != nil {
+		return err
+	}
+
+	wfCloseTime, err := mutableState.GetWorkflowCloseTime(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
What this function returns if the workflow is still running? Don't need to check it first?

Reply:
I believe then the `closeTime` is nil, and it panics when it tries to convert nil to time. I added a check to only get `closeTime` when workflow is not running. Also, in `GetWorkflowCloseTime`, I added error handling for when closeTime is nil, just in case when `processDeleteExecution` is called for workflow with the status "zombie" or "corrupted" and closeTime is nil.

Reply:
removed this code. We are getting workflow closetime from task as mutable state may be deleted when processing delete execution.